service/src/main/java/bio/terra/workspace/service/workspace/flight/GcpCloudSyncStep.java

```diff
@@ -68,17 +88,40 @@ public StepResult doStep(FlightContext flightContext)
               .getIamPolicy(gcpProjectId, new GetIamPolicyRequest())
               .execute();
 
-      List<Binding> newBindings = new ArrayList<>();
-      // Add all existing bindings to ensure we don't accidentally clobber existing permissions.
-      newBindings.addAll(currentPolicy.getBindings());
+      List<Binding> newBindings = new ArrayList<>(currentPolicy.getBindings());
+
       // Add appropriate project-level roles for each WSM IAM role.
       workspaceRoleGroupsMap.forEach(
           (wsmRole, email) -> {
             if (CUSTOM_GCP_PROJECT_IAM_ROLES.containsKey(wsmRole)) {
-              newBindings.add(bindingForRole(wsmRole, email, gcpProjectId));
+              newBindings.add(bindingForRole(wsmRole, GcpUtils.toGroupMember(email), gcpProjectId));
             }
           });
 
+      // Workaround for permission propagation delays: directly grant user and pet the owner
+      // role on the project.
+      if (features.isTemporaryGrantEnabled()) {
+        // Get the user emails we are granting
+        String petMember =
+            GcpUtils.toSaMember(
+                samService.getOrCreatePetSaEmail(gcpProjectId, userRequest.getRequiredToken()));
```

Comment:
The flight working map already has an AuthenticatedUserRequest with the pet email as `WorkspaceFlightMapKeys.PET_SA_CREDENTIALS`, you can read that instead to save a Sam call here

Reply:
Great, thanks for catching that